Source/FieldSolver/SpectralSolver/SpectralFieldData.cpp

```diff
@@ -76,6 +77,13 @@ SpectralFieldIndex::SpectralFieldIndex (const bool update_with_rho,
                 G = c++;
             }
         }
+
+        if (pml_rz)
+        {
+            Er_pml = c++; Et_pml = c++;
+            Br_pml = c++; Bt_pml = c++;
+        }
```

Comment:
Just double checking: since this `if` block is here inside the `if (pml == false)` block (`pml` being the flag for PML in Cartesian geometry), we will set up the memory for a number of spectral fields that is equal to all of those used in Cartesian geometry wihtout PML plus the ones defined here for RZ geometry with PML. Is that correct? If not, we could have a separate `if` block where we store as many as we need. But maybe it's correct as it is, hence my question!

Reply:
This is correct. It will never be the case the both `pml` and 'pml_rz` are true. Also, the RZ pml works differently than Cartesian. With Cartesian, the PML is a completely different set of fields than the main domain. In RZ, the main domain has extra fields added, the ones setup here.